trio/_sync.py

```diff
@@ -84,25 +93,30 @@ def statistics(self):
           :meth:`wait` method.
 
         """
-        return _EventStatistics(tasks_waiting=len(self._tasks))
+        return EventStatistics(tasks_waiting=len(self._tasks))
 
 
 class AsyncContextManagerMixin:
     @enable_ki_protection
-    async def __aenter__(self):
-        await self.acquire()
+    async def __aenter__(self) -> None:
+        await self.acquire()  # type: ignore[attr-defined]
```

Comment:
We can drop this `attr-defined` through fancy self-type schenanigans:

```py
import typing

class ACM(typing.Protocol):
    async def acquire(self) -> None:
        pass

class AsyncContextManagerMixinWithError:
    async def __aenter__(self) -> None:
        await self.acquire()

class AsyncContextManagerMixinWithoutError:
    async def __aenter__(self: ACM) -> None:
        await self.acquire()
```

Feel free to check (I find <https://mypy-play.net> nice for testing small things like this) :D

Reply:
Ooooh, fancy! Leaving it for a future PR, adding a comment with a `#TODO`, as I'm unused to Protocols.